decorators.py

```diff
@@ -0,0 +1,77 @@
+# Core packages
+import functools
+
+# Third party packages
+import flask
+import modules.authentication as authentication
+
+
+def login_required(func):
+    """
+    Decorator that checks if a user is logged in, and redirects
+    to login page if not.
+    """
+    @functools.wraps(func)
+    @private_cache_headers
+    def is_user_logged_in(*args, **kwargs):
+        if not authentication.is_authenticated(flask.session):
+            return flask.redirect('login?next=' + flask.request.path)
+
+        return func(*args, **kwargs)
+    return is_user_logged_in
+
+
+def _cache_headers_decorator(route_function, cache_headers):
+    """
+    Return a decorator function to add a set of Cache-Control to the response
+    """
+
+    @functools.wraps(route_function)
+    def decorated_function(*args, **kwargs):
+        response = flask.make_response(
+            route_function(*args, **kwargs)
+        )
+
+        if response.status_code == 200:
```

Comment:
`if response.ok:`

Reply:
Actually it's worse than that - ["respose.ok: Returns True if status_code is less than 400"](http://docs.python-requests.org/en/master/api/#requests.Response.ok). I *definitely* don't want 3** responses to be cached.